### PR TITLE
Document editorconfig and nixfmt, remove manual rules

### PR DESCRIPTION
**Pull request by infinisil**
## Description of changes

This documents editorconfig and nixfmt in CONTRIBUTING.md. Ping @NixOS/nix-formatting @eclairevoyant 

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -557,138 +557,11 @@ Names of files and directories should be in lowercase, with dashes between words
...
-  but if they don't fit on one line they're written as:
-
-  ```nix
-  { arg1, arg2, arg3
-  , arg4
-  # Some comment...
-  ,  argN
-  }: { }
-  ```
+- New files must be formatted by entering the `nix-shell` from the repository root and running `nixfmt`.
```
Comment (Sigmanificient) [inline CONTRIBUTING.md:564]: I believe edited files are also checked by the CI?
```suggestion
- New or edited files must be formatted by entering the `nix-shell` from the repository root and running `nixfmt`.
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (eclairevoyant) [reply CONTRIBUTING.md:564]: The check is for new and already-formatted files, not all edited files.

Comment (Sigmanificient) [reply CONTRIBUTING.md:564]: yeah, `all Nix files touched by the PR`, maybe `New and Edited` files?

Comment (eclairevoyant) [reply CONTRIBUTING.md:564]: But again, it's not all edited files, nor do we necessarily want people formatting, say,`all-packages.nix` or other unformatted files with a high chance of merge conflicts

Comment (Sigmanificient) [reply CONTRIBUTING.md:564]: Any suggestion then?

Comment (infinisil) [reply CONTRIBUTING.md:564]: As @eclairevoyant also said, it's not all edited ones, see https://github.com/NixOS/nixpkgs/pull/326407 for more context :)

I'd say it's okay to just say "New files" here, for now leaving it undefined how existing files should be formatted. Once we enforce it for all files (probably in like 2 months or so), we can change this to "All files".

Comment (Sigmanificient) [reply CONTRIBUTING.md:564]: Okay, just wanted to be sure, this was mentioned

**Review by Sigmanificient — APPROVED** [review]

**Review by eclairevoyant — APPROVED** [review]
Thank you for documenting the change so quickly!

**Comment by infinisil** [comment]
Thanks for the quick reviews, I see no reason to wait with merging this, let's go! :rocket:

File: CONTRIBUTING.md (hunk 1 above)
Comment (JohnRTitor) [reply CONTRIBUTING.md:564]: Just so we are clear, `nixfmt` in the `nix-shell` is from the `nixfmt-rfc-style` package, right?

**Pull request merged**